client/app/bundles/course/experience-points/disbursement/components/forms/FilterForm.tsx

```diff
@@ -0,0 +1,191 @@
+import { FC, memo } from 'react';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Controller, useForm } from 'react-hook-form';
+import {
+  defineMessages,
+  FormattedMessage,
+  injectIntl,
+  WrappedComponentProps,
+} from 'react-intl';
+import FormDateTimePickerField from 'lib/components/form/fields/DateTimePickerField';
+import FormTextField from 'lib/components/form/fields/TextField';
+import { Button, Grid } from '@mui/material';
+import * as yup from 'yup';
+import formTranslations from 'lib/translations/form';
+import { ForumDisbursementFilters } from 'types/course/disbursement';
+import ErrorText from 'lib/components/ErrorText';
+import { toast } from 'react-toastify';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from 'types/store';
+import { setReactHookFormError } from 'lib/helpers/react-hook-form-helper';
+import { fetchFilteredForumDisbursements } from '../../operations';
+
+interface Props extends WrappedComponentProps {
+  initialValues: ForumDisbursementFilters;
+}
+
+const translations = defineMessages({
+  startTime: {
+    id: 'course.experience-points.disbursement.FilterForm.startTime',
+    defaultMessage: 'Start time   *',
+  },
+  endTime: {
+    id: 'course.experience-points.disbursement.FilterForm.endTime',
+    defaultMessage: 'End time *',
+  },
+  weeklyCap: {
+    id: 'course.experience-points.disbursement.FilterForm.weeklyCap',
+    defaultMessage: 'Weekly Cap',
+  },
+  submit: {
+    id: 'course.experience-points.disbursement.FilterForm.submit',
+    defaultMessage: 'Search',
+  },
+  startEndValidationError: {
+    id: 'course.experience-points.disbursement.ForumDisbursementForm.startEndValidationError',
+    defaultMessage: "Must be after 'Start At'",
+  },
+  fetchFilterSuccess: {
+    id: 'course.experience-points.disbursement.DisbursementForm.fetchFilterSuccess',
+    defaultMessage: 'Successfully filtered forum users.',
+  },
+  fetchFilterNone: {
+    id: 'course.experience-points.disbursement.DisbursementForm.fetchFilterNone',
+    defaultMessage: 'No post made between these 2 dates.',
+  },
+  fetchFilterFailure: {
+    id: 'course.experience-points.disbursement.DisbursementForm.fetchFilterFailure',
+    defaultMessage: 'Failed to retrieve filtered forum users.',
+  },
+});
+
+const FilterForm: FC<Props> = (props) => {
+  const { intl, initialValues } = props;
+  const dispatch = useDispatch<AppDispatch>();
+
+  const validationSchema = yup.object({
+    startDate: yup.date().nullable(),
+    endDate: yup
+      .date()
+      .min(yup.ref('startDate'), translations.startEndValidationError),
+    weeklyCap: yup.string().required(formTranslations.required),
```

Comment:
can add a proper message when it's not a number?
![image](https://user-images.githubusercontent.com/42570513/177941539-7c6b0f13-164a-4eb3-9953-4aed8e7cbcb4.png)